reactive/datanode.py

```diff
@@ -52,6 +61,14 @@ def send_nn_spec(namenode):
     namenode.set_local_spec(bigtop.spec())
 
 
+@when('apache-bigtop-datanode.installed', 'hdfs.storage.available')
+@when_not('apache-bigtop-datanode.started', 'hdfs.storage.prepared')
+def mv_and_symlink_dirs():
+    shutil.move('/var/lib/hadoop-hdfs', '/mnt')
+    os.symlink('/mnt/hadoop-hdfs', '/var/lib/hadoop-hdfs')
```

Comment:
It looks like Bigtop [uses](https://github.com/apache/bigtop/blob/master/bigtop-deploy/puppet/hieradata/site.yaml#L8-L12) `/data/{1..4}` as the default storage location, if I'm reading that right.  Perhaps you could just make the storage target `/data` rather than `/mnt` and not have to do anything else?  Or, if that doesn't work, you should at least be able to override the `hadoop::hadoop_storage_dirs` option directly in the `install_datanode` handler.

All that said, I haven't touched this in a while, so I may be completely wrong here.

Reply:
@johnsca right you are sir!

```bash
$ sudo tree -L 5 /data
/data
├── 1
│   ├── hdfs
│   │   ├── current
│   │   │   ├── BP-49866218-172.31.41.101-1519657520654
│   │   │   │   ├── current
│   │   │   │   ├── scanner.cursor
│   │   │   │   └── tmp
│   │   │   └── VERSION
│   │   └── in_use.lock
│   ├── mapred
│   └── yarn
│       ├── filecache
│       ├── nmPrivate
│       └── usercache
└── 2
    ├── hdfs
    │   ├── current
    │   │   ├── BP-49866218-172.31.41.101-1519657520654
    │   │   │   ├── current
    │   │   │   ├── scanner.cursor
    │   │   │   └── tmp
    │   │   └── VERSION
    │   └── in_use.lock
    ├── mapred
    └── yarn
        ├── filecache
        ├── nmPrivate
        └── usercache

22 directories, 6 files
```